src/librbd/internal.h

```diff
@@ -105,6 +105,8 @@ namespace librbd {
   int list(librados::IoCtx& io_ctx, std::vector<std::string>& names);
   int list_children(ImageCtx *ictx,
 		    std::set<std::pair<std::string, std::string> > & names);
+  int list_children2(ImageCtx *ictx,
```

Comment:
Let's not add a duplicate function within the internal API -- instead, update the existing ```list_children``` method to take ```std::vector<child_info_t>``` and update the public API methods to convert between the new type and the old ```pair```.